### PR TITLE
consistenly check MC fragments version

### PR DESCRIPTION
**Pull request by runcom**
**- What I did**

- simplify the version check for MC fragments (and generated one), we check version anyway for all cause the logic was flawed (`t.versionCheckRequired` wasn't honored in the actual version check if the annotation was there anyway)
- added some comments to the code to clarify why we do something...

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: pkg/operator/status.go [hunk 1]
```diff
@@ -222,31 +222,24 @@ func isMachineConfigPoolConfigurationValid(pool *mcfgv1.MachineConfigPool, versi
...
 
+		// we check that all the machineconfigs (generated, and those that were used to create generated)
+		// were generated by correct version of the controller.
 		v, ok := mc.Annotations[ctrlcommon.GeneratedByControllerVersionAnnotationKey]
-		if t.versionCheckRequired && !ok {
-			return fmt.Errorf("%s must be created by controller version %s", t.name, version)
+		if !ok {
+			return fmt.Errorf("%s must be created by controller version %s", mcName, version)
 		}
 		if ok && v != version {
```
Comment (runcom) [inline pkg/operator/status.go]: `t.versionCheckRequired` was missing here, effectively doing the version check for everything as long as the annotation was there, so it's just irrelevant to have that field around...

Comment (runcom) [reply pkg/operator/status.go]: we set the annotation anyway for _anything_ so there's little to check if the annotation is there

**Comment by runcom** [comment]
build flake

/retest

**Comment by LorbusChris** [comment]
just a note: #471 includes this change in mcc/start.go as well

**Comment by runcom** [comment]
I'll drop that thanks!

**Comment by kikisdeliveryservice** [comment]
Minor nit pr/commit message: "consistenly" -> "consistently"

**Comment by runcom** [comment]
updated and dropped the commit already in #471

**Comment by runcom** [comment]
aws limit :sob:

**Comment by kikisdeliveryservice** [comment]
@runcom we are hitting limits everywhere, gonna elevate the issue in slack.

**Comment by runcom** [comment]
Do we want this for 4.1?

**Comment by LorbusChris** [comment]
I'd say yes.

**Comment by runcom** [comment]
@cgwalters @kikisdeliveryservice @LorbusChris ptal

**Comment by kikisdeliveryservice** [comment]
@runcom just to double check : this is verifying that the controller version (which for example, can now be seen in `oc get machineconfigs`  ->  `4.0.0-alpha.0-9-gfc1a99e6-dirty`) matches the version we expect? and we want to ensure that they are all generated by the same version?

is this to keep track of which MC versions a user is using (in case we change that)? or..?

just trying to understand why we need the check

**Comment by runcom** [comment]
> @runcom just to double check : this is verifying that the controller version (which for example, can now be seen in `oc get machineconfigs` -> `4.0.0-alpha.0-9-gfc1a99e6-dirty`) matches the version we expect? and we want to ensure that they are all generated by the same version?
> 
> is this to keep track of which MC versions a user is using (in case we change that)? or..?
> 
> just trying to understand why we need the check

yeah, the check makes sure every config is generated by the correct version of the operator (operator = (operator, controllers, daemon, server).

Other than the check itself, _this PR isn't changing the behavior_, just simplifying the code w/o using a struct.

**Comment by cgwalters** [comment]
LGTM, giving to

/assign kikisdeliveryservice

for final review and lgtm.

File: pkg/operator/status.go [hunk 2]
```diff
@@ -222,31 +222,30 @@ func isMachineConfigPoolConfigurationValid(pool *mcfgv1.MachineConfigPool, versi
...
 	}
-	for _, t := range tasks {
-		mc, err := machineConfigGetter(t.name)
+	for _, mcName := range mcs {
+		mc, err := machineConfigGetter(mcName)
 		if err != nil {
 			return err
 		}
 
+		// We check that all the machineconfigs (generated, and those that were used to create generated,
```
Comment (kikisdeliveryservice) [inline pkg/operator/status.go]: really really like these comments btw!

**Pull request merged**